substrate/frame/broker/src/lib.rs

```diff
@@ -969,6 +976,16 @@ pub mod pallet {
 			Ok(Pays::No.into())
 		}
 
+		/// Remove a lease.
+		///
+		/// - `origin`: Must be Root or pass `AdminOrigin`.
+		/// - `task`: The workload of the lease which should be removed.
```

Comment:
```suggestion
		/// - `task`: The task id of the lease which should be removed.
```

Also from memory there can be multiple leases for a given `task_id`, although not how this is intended to be used in the system. I don't think it's worth complicating the extrinsic to also check timeslice etc. You should however include in the docs that it removes the first in the vector, which is not necessarily the first to end in time. I'm AFK so can't check right now, but would be good to also cover this behaviour in the test.